### PR TITLE
Feature/invite links

### PR DESCRIPTION
**Pull request by aerych**
Refs #https://github.com/wordpress-mobile/WordPress-iOS/issues/15356

This PR adds API support for invite links.  It includes: 
- Methods for fetching, generating, and disabling invite links
- A new model to represent remote invite links
- Basic unit tests for the new methods.

It's worth mentioning that the data returned from the v1.1 invites and v2.0 generate endpoints differs.  The new remote model relies on default values for fields missing from the generate endpoint. 

@ScoutHarris would you be game for a review? 

### Testing Details

To Test: 
Run unit tests.  Ensure they all pass.
Run the [companion WPiOS PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/15970) to confirm that the new methods work as expected.

- [x] Please check here if your pull request includes additional test coverage.

### CODE REVIEW

File: WordPressKit/PeopleServiceRemote.swift [hunk 1]
```diff
@@ -364,6 +364,97 @@ public class PeopleServiceRemote: ServiceRemoteWordPressComREST {
...
+        }, failure: { (error, httpResponse) in
+            failure(error)
+        })
+    }
+
+    /// Create a new batch of invite links.
+    ///
+    /// - Parameters:
+    ///   - siteID: The site ID for the invite links.
+    ///   - success: A succss block accepting an array of invite links as an argument.
```
Comment (ScoutHarris) [inline WordPressKit/PeopleServiceRemote.swift]: Typo: `succss`

File: WordPressKit/PeopleServiceRemote.swift [hunk 2]
```diff
@@ -364,6 +364,97 @@ public class PeopleServiceRemote: ServiceRemoteWordPressComREST {
             failure(error)
         })
     }
+
+    /// Fetch any existing invite links.
+    ///
+    /// - Parameters:
+    ///   - siteID: The site ID for the invite links.
+    ///   - success: A succss block accepting an array of invite links as an argument.
```
Comment (ScoutHarris) [inline WordPressKit/PeopleServiceRemote.swift]: Typo: `succss`

File: WordPressKit/PeopleServiceRemote.swift [hunk 3]
```diff
@@ -364,6 +364,97 @@ public class PeopleServiceRemote: ServiceRemoteWordPressComREST {
...
+            failure(error)
+        })
+
+    }
+
+    /// Disable any existing invite links.
+    ///
+    /// - Parameters:
+    ///   - siteID: The site ID for the invite links to disable.
+    ///   - success: A succss block.
```
Comment (ScoutHarris) [inline WordPressKit/PeopleServiceRemote.swift]: Typo: `succss`

**Review by ScoutHarris — APPROVED** [review]
Other than a few typos, looks good!

:shipit:

**Comment by ScoutHarris** [comment]
Oh, you'll need to bump the version since I just used `beta.3`. Sorry!

**Comment by aerych** [comment]
Thanks @ScoutHarris !

**Pull request merged**